### PR TITLE
Remove alchy from cg store

### PR DESCRIPTION
**Pull request by seallard**
## Description
One goal of the refactoring is to enable removing alchy, which in turn will enable us to update the sqlalchemy package.
This PR applies the necessary changes in order to remove Alchy as a dependency in the cg store.

Potential issues:
- The alchy declarative_base model included a default to_dict method: check all instances of to_dict
- SqlAlchemy requires each table to be given an explicit name - some might differ from the existing ones:  compare table names in the existing databases.

### Changed
Replace all instances of
- `add_commit` with `session.add` followed by `session.commit`
- `delete_commit` with `session.delete` followed by `session.commit`
- `commit` with `session.commit`
- `delete` with `session.delete`
- `from alchy import Query` with `from sqlalchemy.orm import Query`
- `rollback` with `session.rollback`
- `close` with `session.close`
- `add` with multiple objects with `add_all`

Update the model definitions
- All cg models now inherit from `sqlalchemy.ext.declarative` instead of `alchy.make_declarative`
- Each table has its name explicitly specified with the `__tablename__` attribute

The store is now instantiated with a session, and all interactions with the db go through the session.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by

### This [version](https://semver.org/) is a
- [ ] **MINOR** - when you add functionality in a backwards compatible manner

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/microsalt.py [hunk 1]
```diff
@@ -164,7 +164,9 @@ def get_samples(self, case_id: str, sample_id: Optional[str] = None) -> List[Sam
         If sample_id is specified, will return a list with only this sample_id.
         Otherwise, returns all samples in given case"""
         if sample_id:
-            return [self.status_db.query(Sample).filter(Sample.internal_id == sample_id).first()]
+            return [
```
Comment (ChrOertlin) [inline cg/meta/workflow/microsalt.py:167]: Would that work here?
```suggestion
            return [self.status_db.session.get_sample_by_internal_id(internal_id=sample_id)
```

File: cg/meta/workflow/microsalt.py [hunk 2]
```diff
@@ -164,7 +164,9 @@ def get_samples(self, case_id: str, sample_id: Optional[str] = None) -> List[Sam
...
+                self.status_db.session.query(Sample).filter(Sample.internal_id == sample_id).first()
+            ]
 
         case_obj: Family = self.status_db.get_case_by_internal_id(internal_id=case_id)
         return [link.sample for link in case_obj.links]
```
Comment (ChrOertlin) [inline cg/meta/workflow/microsalt.py:172]: does this also require a .session?

File: cg/meta/workflow/microsalt.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/meta/workflow/microsalt.py:167]: likely out of scope, just wondering!

File: cg/store/api/add.py [hunk 3]
```diff
@@ -63,12 +64,12 @@ def add_customer(
     def add_collaboration(self, internal_id: str, name: str, **kwargs) -> Collaboration:
         """Build a new customer group record."""
 
-        return self.Collaboration(internal_id=internal_id, name=name, **kwargs)
+        return Collaboration(internal_id=internal_id, name=name, **kwargs)
```
Comment (ChrOertlin) [inline cg/store/api/add.py:67]: for these functions we can then remove the self parameter in the definition, is this correct?

File: cg/meta/workflow/microsalt.py (hunk 2 above)
Comment (seallard) [reply cg/meta/workflow/microsalt.py:172]: No

File: cg/store/api/add.py (hunk 3 above)
Comment (seallard) [reply cg/store/api/add.py:67]: If we want to make them static, yes

File: cg/meta/workflow/microsalt.py (hunk 1 above)
Comment (seallard) [reply cg/meta/workflow/microsalt.py:167]: Session is not required, created a PR for it here https://github.com/Clinical-Genomics/cg/pull/1946

File: cg/meta/workflow/microsalt.py (hunk 2 above)
Comment (seallard) [reply cg/meta/workflow/microsalt.py:172]: It is only needed when we interact directly with the database

**Comment by seallard** [comment]
Searching the codebase with the regex pattern

- `(?<!session\.)\bcommit\(`: fine
- `(?<!session\.)\bdelete\(`: fine
- `(?<!session\.)\brollback\(`: fine
- `(?<!session\.)\bclose\(`: fine

File: cg/store/api/add.py [hunk 4]
```diff
@@ -208,7 +209,7 @@ def relate_sample(
     ) -> FamilySample:
         """Relate a sample record to a family record."""
 
-        new_record = self.FamilySample(status=status)
+        new_record = FamilySample(status=status)
```
Comment (henrikstranneheim) [inline cg/store/api/add.py:212]: Add type hint

File: cg/store/api/add.py [hunk 5]
```diff
@@ -264,7 +265,7 @@ def add_panel(
     ) -> Panel:
         """Build a new panel record."""
 
-        new_record = self.Panel(
+        new_record = Panel(
```
Comment (henrikstranneheim) [inline cg/store/api/add.py:268]: Add type hint

File: cg/store/api/add.py [hunk 6]
```diff
@@ -280,21 +281,19 @@ def add_pool(
...
-        capture_kit: str = None,
         invoice_id: int = None,
         no_invoice: bool = None,
         delivered_at: dt.datetime = None,
     ) -> Pool:
         """Build a new Pool record."""
 
-        new_record = self.Pool(
+        new_record = Pool(
             name=name,
```
Comment (henrikstranneheim) [inline cg/store/api/add.py:291]: Add type hint

File: cg/store/api/add.py [hunk 7]
```diff
@@ -314,7 +313,7 @@ def add_delivery(
 
         if not any([sample, pool]):
             raise ValueError("you have to provide a sample or a pool")
-        new_record = self.Delivery(destination=destination, comment=comment)
+        new_record = Delivery(destination=destination, comment=comment)
         new_record.sample = sample
```
Comment (henrikstranneheim) [inline cg/store/api/add.py:317]: Type hint

File: cg/store/api/add.py [hunk 8]
```diff
@@ -333,7 +332,7 @@ def add_invoice(
         """Build a new Invoice record."""
 
         new_id = self.new_invoice_id()
-        new_invoice = self.Invoice(
+        new_invoice = Invoice(
             comment=comment,
```
Comment (henrikstranneheim) [inline cg/store/api/add.py:336]: Type hint

File: cg/store/models.py [hunk 9]
```diff
@@ -19,7 +19,14 @@
 
 from cg.constants.constants import CONTROL_OPTIONS, PrepCategory
 
-Model = alchy.make_declarative_base(Base=alchy.ModelBase)
+Model = declarative_base()
+
+
+def to_dict(model_instance):
+    if hasattr(model_instance, "__table__"):
+        return {c.name: getattr(model_instance, c.name) for c in model_instance.__table__.columns}
```
Comment (henrikstranneheim) [inline cg/store/models.py:27]:
```suggestion
        return {column.name: getattr(model_instance, column.name) for column in model_instance.__table__.columns}
```

**Comment by seallard** [comment]
Spend some extra time reviewing these files:
- [cg/store/api/base.py](https://github.com/Clinical-Genomics/cg/pull/1945/files#diff-c4ec034b797fa1b066e34aaf106926780be815e237c6e28ea3682363ce32a259)
- [cg/server/ext.py](https://github.com/Clinical-Genomics/cg/pull/1945/files#diff-c6f8cc84eb12e4f79caff9a00c25b27c2c0f8adbedffe4b90137f02f301f0bb5)
- [cg/store/api/core.py](https://github.com/Clinical-Genomics/cg/pull/1945/files#diff-447ef08e4d3d2ed9f598891c60e50b3b55e723adeefe7aee8b737c2bd347d653)

File: tests/cli/test_base.py [hunk 10]
```diff
@@ -32,36 +32,36 @@ def test_missing_command(cli_runner: CliRunner):
     assert result.exit_code != 0
 
 
-def test_cli_init(cli_runner: CliRunner, base_context: CGConfig, caplog):
+def test_cli_init(cli_runner: CliRunner, base_context: CGConfig, caplog, tmp_path: Path):
```
Comment (henrikstranneheim) [inline tests/cli/test_base.py:35]: Well done 👍

**Review by henrikstranneheim — APPROVED** [review]
Fantastic 🚀🎉

File: tests/cli/test_base.py (hunk 10 above)
Comment (seallard) [reply tests/cli/test_base.py:35]: Vincent helped me out 🥇

**Comment by seallard** [comment]
Deployed and tested in stage. status-db seems to work as expected, same for the order portal and the cli tool on Hasta.

**Comment by seallard** [comment]
We probably want to add some serialisation library such as `sqlalchemy-serialize`

**Comment by seallard** [comment]
Deployed to stage and prod.

**Pull request merged**